### PR TITLE
Update envoy SHA to dfec97814ae912d5b3188e85be400dbd343cdbf0

### PR DESCRIPTION
**Pull request by nrjpoddar**

### CODE REVIEW

**Comment by rlenglet** [comment]
@nrjpoddar Please add an item in the [1.3.3 release note draft](https://github.com/istio/istio/wiki/Release-Note-1.3.3-Draft) for https://github.com/istio/envoy/pull/109 in layman terms.

**Comment by nrjpoddar** [comment]
> @nrjpoddar Please add an item in the [1.3.3 release note draft](https://github.com/istio/istio/wiki/Release-Note-1.3.3-Draft) for [istio/envoy#109](https://github.com/istio/envoy/pull/109) in layman terms.

yes, thanks for reminding and making sure it's in layman lingo. Cheers! :)

**Comment by nrjpoddar** [comment]
@rlenglet took a stab at the Release Notes: https://github.com/istio/istio/wiki/Release-Note-1.3.3-Draft

Feel free to augment it if it's still unclear or too complex.

**Comment by rlenglet** [comment]
That looks great, thank!

**Comment by rlenglet** [comment]
We can close this PR. https://github.com/istio/proxy/pull/2455 will be merge, which picks up a more recent `istio/envoy:release-1.3` SHA.
